apps/web/src/components/auth/LoginForm.tsx

```diff
@@ -58,9 +60,21 @@ export function LoginForm({ email, token }: Props) {
 
     try {
       const response = await mutateAsync(itemData);
-      setToken((response as any).token);
-      if (isFromVercel) return;
-      if (!token) navigate('/templates');
+      const token = (response as any).token;
+      if (isFromVercel) {
+        setToken(token);
+
+        return;
+      }
+
+      if (invitationToken) {
+        submitToken(token, invitationToken);
+
+        return;
+      }
```

Comment:
The LoginForm component is used on the: Login page and Invitation Page, when it's used on the second one the `invitationToken` is passed, and then when the "Sign In & Accept" button is pressed we do submit the token.